packages/b/bgfx/rules/shaders.lua

```diff
@@ -38,26 +45,7 @@ rule("shaders")
         if fileconfig and fileconfig.vardef then
             vardef_filename = fileconfig.vardef
         else
-            vardef_filename = path.join(
-                path.directory(shaderfile),
-                path.basename(shaderfile) .. ".varying.def.sc")
-        end
-
-        local shader_type
-        if fileconfig and fileconfig.type then
-            if table.contains(bgfx_types, fileconfig.type) then
-                shader_type = fileconfig.type
-            else
-                raise("unsupported shader type " .. fileconfig.type)
-            end
-        elseif shaderfile:match("%.vert$") then
-            shader_type = "vertex"
-        elseif shaderfile:match("%.frag$") then
-            shader_type = "fragment"
-        elseif shaderfile:match("%.comp$") then
-            shader_type = "compute"
-        else
-            raise("cannot determine shader type from file name " .. path.filename(shaderfile))
+            vardef_filename = path.join(path.directory(shaderfile), "varying.def.sc")
```

Comment:
Will it conflict when compiling multiple shaders in parallel?

Reply:
I followed the [shaderc behavior](https://github.com/bkaradzic/bgfx/blob/eaec8ddf3b4c5a11e60cde42728439293721ec8d/tools/shaderc/shaderc.cpp#L2827) which uses the file varying.def.sc in the shader directory.